rpc/src/client.rs

```diff
@@ -0,0 +1,18 @@
+use http::{header, HeaderValue};
+use jsonrpsee::http_client::{HeaderMap, HttpClient, HttpClientBuilder};
+
+pub fn new_http(conn_str: &str, auth_token: Option<&str>) -> crate::Result<HttpClient> {
+    let headers: HeaderMap<HeaderValue> = auth_token
+        .into_iter()
+        .map(|token| {
+            HeaderValue::from_str(&format!("Bearer {token}"))
+                .map(|value| (header::AUTHORIZATION, value))
+        })
+        .collect::<Result<_, _>>()?;
```

Comment:
For me this is much more cleaner
```suggestion
    let mut headers = HeaderMap::new();

    if let Some(token) = auth_token {
        let val = HeaderValue::from_str(&format!("Bearer {token}"))?;
        headers.insert(header::AUTHORIZATION, val);
    }
```